src/lib/net/SecureSocket.cpp

```diff
@@ -699,19 +667,30 @@ SecureSocket::disconnect()
     sendEvent(getEvents()->forIStream().inputShutdown());
 }
 
-bool SecureSocket::verify_cert_fingerprint(const barrier::fs::path& fingerprint_db_path)
+bool SecureSocket::verify_peer_certificate(const barrier::fs::path& fingerprint_db_path)
 {
     // ssl_mutex_ is assumed to be acquired
 
+    // ensure peer presented a certificate
+    X509* cert = SSL_get_peer_certificate(m_ssl->m_ssl);
+    if (cert == NULL) {
```

Comment:
nullptr

Reply:
Sure. I was trying to follow the majority of other parts of the file.. Though there are both NULL and nullptr used. I must say it is not all that consistent code style in existing code, while you have rather specific expectations for new code, which is kind of hard to know in advance..

Reply:
> while you have rather specific expectations for new code, which is kind of hard to know in advance..

I'm not blaming you, it's impossible to know all expectations. I'm just pointing out the better option, let's fix it efficiently and move on :-)